include/swift/AST/DiagnosticsSema.def

```diff
@@ -942,6 +942,13 @@ ERROR(need_cxx_interop_to_import_module,none,
 NOTE(enable_cxx_interop_docs,none,
      "visit https://www.swift.org/documentation/cxx-interop/project-build-setup to learn how to enable C++ interoperability", ())
 
+ERROR(cxx_stdlib_kind_mismatch,none,
+      "module %0 was built with "
+      "%select{platform default C++ stdlib|libc++}1, "
+      "but current compilation uses "
+      "%select{platform default C++ stdlib|libc++}2",
```

Comment:
I'm not sure that this is great - the platform default c++ runtime could be libc++. I'd rather that we use a string parameter and let the frontend specify the runtime name.

Reply:
I changed this to explicitly list the supported C++ stdlibs by name.